RealmSwift/Tests/SwiftTestObjects.swift

```diff
@@ -481,3 +481,36 @@ class SwiftCircleObject: Object {
     @objc dynamic var obj: SwiftCircleObject?
     let array = List<SwiftCircleObject>()
 }
+
+class SwiftGenericPropsOrderingParent: Object {
+    let parentFirstList = List<SwiftIntObject>()
+    @objc dynamic var parentFirstNumber = 0
+    func parentFunction() -> Int { return parentFirstNumber + 1 }
+    @objc dynamic var parentSecondNumber = 1
+    var parentComputedProp: String { return "hello world" }
+}
+
+class SwiftGenericPropsOrderingObject: SwiftGenericPropsOrderingParent {
```

Comment:
Other things that would be good to test:

1. A `@objc dynamic` ignored property.
2. A `@objc dynamic` computed property.
3. A lazy property
4. A `RealmOptional` property